src/pages/workspaces/workspace/Notebooks.js

```diff
@@ -148,22 +148,29 @@ class NotebooksContent extends Component {
   async uploadFiles(files) {
     const { namespace, workspace: { workspace: { bucketName } } } = this.props
     const existingNames = this.getExistingNames()
-    try {
-      this.setState({ saving: true })
-      await Promise.all(_.map(async file => {
-        const name = file.name.slice(0, -6)
-        if (_.includes(name, existingNames)) {
-          throw new Error(`${name} already exists`)
+    this.setState({ saving: true })
+    await Promise.all(_.map(async file => {
+      const name = file.name.slice(0, -6)
+      if (_.includes(name, existingNames)) {
+        let c = 1
+        let doubleName = true
+        let NewName = `${name}(${c})`
+        while (doubleName) {
+          if (_.includes(NewName, existingNames)) {
+            c += 1
+            NewName = `${name}(${c})`
+          } else {
+            doubleName = false
+          }
```

Comment:
This can get collapsed down quite a bit. Start with

    ...
    const name = file.name.slice(0, -6)
    let resolvedName = name
    let c = 0
    while (_.includes(resolvedName, existingNames)) {
      resolvedName = `${name}(${++c})`
    }

Then you can just have the Utils.readFileAsText and notebook create lines below once.

Reply:
BTW, this is the kind of thing that makes me really miss CLJS.
```clojure
(->> (range)
     rest
     (map #(str filename "(" % ")"))
     (remove existing-names)
     first)
```